modules/services/git-workspace.nix

```
@@ -0,0 +1,153 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.git-workspace;
+  tomlFormat = pkgs.formats.toml { };
+  workspaceType = lib.types.submodule {
+    freeformType = tomlFormat.type;
+    options.provider = lib.mkOption {
+      type = lib.types.listOf (lib.types.submodule {
+        freeformType = tomlFormat.type;
+        options = {
+          provider = lib.mkOption {
+            type = lib.types.str;
+            example = "github";
+          };
+          name = lib.mkOption {
```

Please separate options by empty lines to help make things a bit more readable:

```suggestion

          name = lib.mkOption {
```